packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain/useMaxAmount.ts

```diff
@@ -33,20 +33,28 @@ export function useMaxAmount() {
       return undefined
     }
 
-    // For custom fee token deposits, we can set the max amount, as the fees will be paid in ETH
-    if (nativeCurrency.isCustom && isDepositMode) {
-      return utils.formatUnits(
-        nativeCurrencySourceBalance,
-        nativeCurrencyDecimalsOnSourceChain
-      )
-    }
-
     // ETH deposits and ETH/custom fee token withdrawals
     const nativeCurrencyBalanceFormatted = utils.formatUnits(
       nativeCurrencySourceBalance,
       nativeCurrencyDecimalsOnSourceChain
     )
 
+    if (nativeCurrency.isCustom && isDepositMode) {
+      if (selectedToken) {
+        // For custom fee token deposits, we can set the max amount, as the fees will be paid in native-currency
+        return utils.formatUnits(
+          nativeCurrencySourceBalance,
+          nativeCurrencyDecimalsOnSourceChain
+        )
+      }
+
+      // for custom fee native token deposits, we will need to subtract the child gas fee from source-balance
+      return String(
+        parseFloat(nativeCurrencyBalanceFormatted) -
+          (estimatedChildChainGasFees ?? 0) * 1.4
```

Comment:
nit: we can move this entire block after the `if` below so we don't have to `??`